google-cloud-firestore/acceptance/firestore/collection_group_test.rb

```diff
@@ -179,5 +179,33 @@
       end
       _(result).must_equal results[1]
     end
+
+    it "queries a collection group using partitions on different timestamps" do
+      rand_col = firestore.col "#{root_path}/query/#{SecureRandom.hex(4)}"
+
+      read_time = Time.now
+      sleep(1)
+
+      document_ids = ["a", "b", "c"].map do |prefix|
+        128.times.map do |i|
+          "#{prefix}#{(i+1).to_s.rjust(3, '0')}"
+        end
+      end.flatten # "a001", "a002", ... "c128"
+      firestore.batch do |b|
+        document_ids.each do |id|
+          doc_ref = rand_col.document id
+          b.set doc_ref, { foo: id }
+        end
+      end
+
+      collection_group = firestore.collection_group(rand_col.collection_id)
+
+      partitions = collection_group.partitions 6, read_time: read_time
```

Comment:
What if there's some time skew between the client and server? Will this test work or will it flake?